libs/feature/search/src/lib/utils/mapper/elasticsearch.field.mapper.ts

```diff
@@ -199,15 +199,26 @@ export class ElasticsearchFieldMapper {
   }
 
   mapLink(sourceLink: SourceWithUnknownProps): MetadataLink | null {
-    const url = getAsUrl(selectField<string>(sourceLink, 'url'))
+    let url
+    let name
+    let description
+    if (sourceLink.urlObject) {
+      url = sourceLink.urlObject['default'] as string
+      name = sourceLink.nameObject['default']
+      description = sourceLink.descriptionObject['default']
+    } else {
+      url = getAsUrl(selectField<string>(sourceLink, 'url'))
+      name = selectField<string>(sourceLink, 'name')
+      description = selectField<string>(sourceLink, 'description')
+    }
```

Comment:
I think you can have a slightly simpler implementation here using `selectTranslatedValue`. For instance:

```js
const url = getAsUrl(
  selectFallback(
    selectTranslatedField<string>(sourceLink, 'urlObject'),
    selectField<string>(sourceLink, 'url')
  )
)
```

and same for `name` and `description`. Relying on the functions from `atomic-operations` make it safer and less verbose to parse the ES object.

Reply:
Did not know about `selectFallBack` and `selectTranslatedField`, usefull ones! and that's done!